### PR TITLE
layers: Remove unused includes

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Installed the new Zed editor for Linux, tried it out, found some unused `#include` and spent some time removing them to hopefully improve compiling speed a bit (and keep things more tidy)

### CODE REVIEW

File: layers/core_checks/cc_drawdispatch.cpp [hunk 1]
```diff
@@ -1738,4 +1737,4 @@ bool CoreChecks::ValidateDrawDualSourceBlend(const LastBound &last_bound_state,
     }
 
     return skip;
-}
\ No newline at end of file
+}
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_drawdispatch.cpp:1740]: idk why git displays it as this, but @arno-lunarg seems Zed enforces the new-line at the end for you

File: layers/state_tracker/device_memory_state.cpp [hunk 2]
```diff
@@ -373,4 +372,3 @@ void vvl::Bindable::CacheInvalidMemory() const {
         }
     }
 }
-
```
Comment (spencer-lunarg) [inline layers/state_tracker/device_memory_state.cpp:376]: even removes when there are 2 new lines at the end to just 1

File: layers/state_tracker/shader_stage_state.h [hunk 3]
```diff
@@ -17,7 +17,8 @@
  */
 
 #pragma once
-#include "utils/vk_layer_utils.h"
```
Comment (spencer-lunarg) [inline layers/state_tracker/shader_stage_state.h:20]: didn't have time, but we are bad at relying on things in `vk_layer_utils.h` to be consumed everywhere... makes touching that file awful as you have to rebuild everything

File: layers/core_checks/cc_drawdispatch.cpp (hunk 1 above)
Comment (arno-lunarg) [reply layers/core_checks/cc_drawdispatch.cpp:1740]: <proceeds to wipe out windows and immediately install linux with this code editor>

**Review by arno-lunarg — APPROVED** [review]

**Pull request merged**